image/preseed/reset.go

```diff
@@ -135,37 +135,39 @@ func ResetPreseededChroot(preseedChroot string) error {
 	// e.g.
 	// lxd.lxc -> /snap/core/current/usr/lib/snapd/complete.sh
 	// lxc -> lxd.lxc
-	files, err := ioutil.ReadDir(filepath.Join(preseedChroot, dirs.CompletersDir))
-	if err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("error reading %s: %v", dirs.CompletersDir, err)
-	}
-	completeShSymlinks := make(map[string]string)
-	var otherSymlinks []string
-
-	// pass 1: find all symlinks pointing at complete.sh
-	for _, fileInfo := range files {
-		if fileInfo.Mode()&os.ModeSymlink == 0 {
-			continue
+	for _, completersPath := range []string{dirs.CompletersDir, dirs.LegacyCompletersDir} {
```

Comment:
Hmm perhaps @stolowski has some insight here, but AFAIU this is called when cleaning up a system image when it was preseeded by the current snapd version. Much of the trouble before was due to snapd writing under the real /usr/share/bas-completion, which we don't do now. Thus, if the current version only writes all the files files under dirs.CompletersDir now, the whole process becomes much simpler now as that directory will already get removed, and there's nothing to do?

Reply:
Yes, I think (AFAIU) we can just add `CompletersDir` to the list of directories removed recursively (leaving the top dir intact) above; I think though that we need to keep existing logic to clean `dirs.LegacyCompletersDir` in case a new version of `snap-preseed` command is run against older snapd?